python/tools/recorded-call-graph-metrics/src/cg_trace/utils.py

```diff
@@ -0,0 +1,21 @@
+def better_compare_for_dataclass(cls):
+    """When dataclass is used with `order=True`, the comparison methods is only implemented for
+    objects of the same class. This decorator extends the functionality to compare class
+    name if used against other objects.
+    """
+    for op in [
+        "__lt__",
+        "__le__",
+        "__gt__",
+        "__ge__",
+    ]:
+        old = getattr(cls, op)
+
+        # Fix loop variable capture (py/loop-variable-capture)
```

Comment:
I'm not sure I understand the function of this comment. It looks like a note to self?

Reply:
yeah, I'm not sure why that got committed in the first place